controllers/resource_helper.go

```diff
@@ -61,36 +73,170 @@ func buildContainerSpec(instance *llamav1alpha1.LlamaStackDistribution, image st
+func addCABundleVolumeMount(instance *llamav1alpha1.LlamaStackDistribution, container *corev1.Container) {
+	if instance.Spec.Server.TLSConfig != nil && instance.Spec.Server.TLSConfig.CABundle != nil {
+		caBundleKey := getCABundleKey(instance.Spec.Server.TLSConfig.CABundle)
+
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:      "ca-bundle",
+			MountPath: "/etc/ssl/certs/" + caBundleKey,
+			SubPath:   caBundleKey,
+			ReadOnly:  true,
+		})
 	}
+}
 
-	if len(instance.Spec.Server.ContainerSpec.Command) > 0 {
-		container.Command = instance.Spec.Server.ContainerSpec.Command
+// getCABundleKey returns the effective CA bundle key name for volume mounting.
+// When multiple keys are specified, we use "ca-bundle.crt" as the consolidated filename.
+func getCABundleKey(caBundleConfig *llamav1alpha1.CABundleConfig) string {
+	// If multiple keys are specified, use a standard filename for the consolidated bundle
+	if len(caBundleConfig.ConfigMapKeys) > 0 {
+		return "ca-bundle.crt"
 	}
 
-	if len(instance.Spec.Server.ContainerSpec.Args) > 0 {
-		container.Args = instance.Spec.Server.ContainerSpec.Args
+	// Use legacy single key behavior
+	if caBundleConfig.ConfigMapKey != "" {
+		return caBundleConfig.ConfigMapKey
+	}
+
+	// Default fallback
+	return defaultCABundleKey
+}
+
+// createCABundleVolume creates the appropriate volume configuration for CA bundles.
+// For single key: uses direct ConfigMap volume.
+// For multiple keys: uses emptyDir volume with InitContainer to concatenate keys.
+func createCABundleVolume(caBundleConfig *llamav1alpha1.CABundleConfig) corev1.Volume {
+	// For multiple keys, we'll use an emptyDir that gets populated by an InitContainer
+	if len(caBundleConfig.ConfigMapKeys) > 0 {
+		return corev1.Volume{
+			Name: "ca-bundle",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		}
+	}
+
+	// For single key (legacy behavior), use direct ConfigMap volume
+	return corev1.Volume{
+		Name: "ca-bundle",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: caBundleConfig.ConfigMapName,
+				},
+			},
+		},
+	}
+}
+
+// createCABundleInitContainer creates an InitContainer that concatenates multiple CA bundle keys
+// from a ConfigMap into a single file in the shared ca-bundle volume.
+func createCABundleInitContainer(caBundleConfig *llamav1alpha1.CABundleConfig) corev1.Container {
+	// Build the shell command to concatenate all specified keys
+	keyPaths := make([]string, 0, len(caBundleConfig.ConfigMapKeys))
+
+	for _, key := range caBundleConfig.ConfigMapKeys {
+		keyPaths = append(keyPaths, "/tmp/ca-source/"+key)
+	}
+
+	// Command to concatenate all keys into the target file
+	command := fmt.Sprintf("cat %s > /etc/ssl/certs/ca-bundle.crt", strings.Join(keyPaths, " "))
```

Comment:
This is dangerous, we are interpolating user-controlled (keyPaths) directly into a shell command string. This risks a shell-injection attack. 

We must do a bit more validation on `keyPaths` first by escaping shell metacharacters. Or use a library like https://github.com/alessio/shellescape